### PR TITLE
Convert useGroupCall to TS

### PR DESCRIPTION
**Pull request by dbkr**

### CODE REVIEW

File: src/room/useGroupCall.ts [hunk 1]
```diff
@@ -18,10 +18,57 @@ import { useCallback, useEffect, useState } from "react";
...
+  GroupCall,
 } from "matrix-js-sdk/src/webrtc/groupCall";
+import { MatrixCall } from "matrix-js-sdk/src/webrtc/call";
+import { CallFeed } from "matrix-js-sdk/src/webrtc/callFeed";
+import { RoomMember } from "matrix-js-sdk/src/models/room-member";
+
 import { usePageUnload } from "./usePageUnload";
 
-export function useGroupCall(groupCall) {
+export interface UseGroupCallType {
```
Comment (dbkr) [inline src/room/useGroupCall.ts:29]: If anyone has any ideas or knows of conventions on what to call this type, I'd love it to have a better name.

File: src/room/useGroupCall.ts [hunk 2]
```diff
@@ -41,20 +88,25 @@ export function useGroupCall(groupCall) {
...
+    localScreenshareFeed: null,
+    localDesktopCapturerSourceId: null,
     requestingScreenshare: false,
     participants: [],
     hasLocalParticipant: false,
   });
 
-  const updateState = (state) =>
+  const updateState = (state: Partial<State>) =>
     setState((prevState) => ({ ...prevState, ...state }));
```
Comment (robintown) [inline src/room/useGroupCall.ts:110]: Given that all updates to state happen through this function, this really feels to me like a case where we should just use individual state hooks

**Review by robintown — APPROVED** [review]
Looks good either way

File: src/room/useGroupCall.ts (hunk 1 above)
Comment (robintown) [reply src/room/useGroupCall.ts:29]: `GroupCallInfo` is all I can think of

File: src/room/useGroupCall.ts (hunk 2 above)
Comment (dbkr) [reply src/room/useGroupCall.ts:110]: Yeah, or a least a few different grouped ones.

**Pull request merged**